core/core.go

```diff
@@ -535,11 +552,23 @@ func Decrypt(jsonIn []byte) ([]byte, error) {
 		return jsonStatusError(err)
 	}
 
-	data, names, secure, err := crypt.Decrypt(s.Data, s.Name)
+	data, names, usages, secure, err := crypt.Decrypt(s.Data, s.Name)
 	if err != nil {
 		return jsonStatusError(err)
 	}
 
+	// a file must be marked as usable for "decrypt" before we will decrypt
+	// it for the user. If no usages are provided, we permit decryption
+	foundDecrypt := false
+	for _, usage := range usages {
+		if usage == "decrypt" {
+			foundDecrypt = true
```

Comment:
Trivial optimization: You can break here.  You can also wrap this in a if len(usages) != 0 instead of check below.

usages is sorted by Encrypt, right?  What about sort.SearchStrings?